staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -1732,6 +1732,11 @@ type VolumeMount struct {
 	// This field is beta in 1.10.
 	// +optional
 	MountPropagation *MountPropagationMode `json:"mountPropagation,omitempty" protobuf:"bytes,5,opt,name=mountPropagation,casttype=MountPropagationMode"`
+	// Expanded path within the volume from which the container's volume should be mounted.
+	// Defaults to "" (volume's root).
+	// SubPathExpr and SubPath are mutually exclusive.
+	// +optional
```

Comment:
Did you consider making this a pointer for extra clarity about whether the user intended to set it?

Reply:
You can also consider making SubPath a pointer. That shouldn't be a breaking change in the wire format.

Reply:
I wanted to keep subPathExpr in the same style and context as subPath. I really dont want to alter subPath behaviour as it could be potentially risky and is also not in the scope of the PR.

Reply:
@liggitt do all new fields need to be pointers?  I vaguely recall something about breaking older clients.

Reply:
New fields must be optional and omitempty, and not result in any serialized output when unspecified. For strings, either `string` or `*string` can meet those requirements

If you need to be able to distinguish between an explicitly specified zero value and an omitted value, `*string` is required. That doesn't seem to be the case here.

In general, for optional fields that get defaults applied to them, making them pointers and only defaulting the nil value is recommended. It's somewhat confusing for an explicit `foo: ""` to get changed to `foo: "some default"` by defaulting.

In this case, since there's no defaulting, and `string` works with `omitempty`, and `subPath` already exists, I'd probably match `subPath`